### ISSUE TITLE
Nested routes within AutoTabsScaffold

### ISSUE THREAD

**Issue by scottbisaillon** [Milad-Akarie/auto_route_library#661]
Currently I have a list view inside a tab. When clicking on an item the user is navigated to a details page. My desire is the details page's scaffold will replace that of the list views page scaffold, however, I am seeing the following:

![Screenshot_20210821-060033.jpg](https://user-images.githubusercontent.com/21369551/130318467-8ddfd200-ee49-4117-b23e-2db85945307c.jpg)

![Screenshot_20210821-060038.jpg](https://user-images.githubusercontent.com/21369551/130318530-14715549-8383-4c83-9f5f-72b1fdb5991e.jpg)

The details view is wrapped by the parent `AppBar` and `BottomNavigationBar`. 

For reference, here is how I am setting my routes:

```dart
@MaterialAutoRouter(
  routes: <AutoRoute>[
    AutoRoute(
      page: RecipeTrackerHomePage,
      path: '/',
      children: [
        AutoRoute(
          path: 'recipes',
          name: 'RecipesRouter',
          page: EmptyRouterPage,
          children: [
            AutoRoute(path: '', page: RecipesPage),
            AutoRoute(path: ':recipeId', page: RecipeDetailsPage),
            AutoRoute<List<IngredientModel>>(
                path: 'add-ingredients', page: AddIngredientPage),
            AutoRoute<List<RecipeStepModel>>(
                path: 'add-recipe-steps', page: AddRecipeStepPage),
            AutoRoute(path: 'new-recipe', page: NewRecipePage),
          ],
        ),
        AutoRoute(path: 'recipes/favorites', page: FavoriteRecipesPage),
      ],
    ),
  ],
  preferRelativeImports: false,
)
class $AppRouter {}
```

**Comment by gerricom** [comment]
> Here is a minimal example of the behavior I am observing. In the sample project, my desire would be to have TabOneChildPage to be a full screen (have its own Scaffold) rather than be wrapped by the Scaffold of HomePage.

I'm fiddling around with this at the moment and as far as I understand this is kind of expected behaviour. Your `home_page.dart` implements an `AutoTabsScaffold` that has a global `AppBar`. The navigation itself is happening in the body of the `AutoTabScaffold`. If you push a page with it's own `AppBar` you'll push it only to the body of the `AutoTabsScaffold`.
If you don't want this, you remove the `appBar` from the `AutoTabsScaffold` and wrap every tab and each subpage with it's own Scaffold. This then requires of course that you properly build every page with an AppBar otherwise your users will miss an UI for navigating back.

**Comment by a7md0** [comment]
I have been looking into this too, I've already removed the `appBarBuilder` from `AutoTabsScaffold` and added new `Scaffold` in all pages with their own `appBar`. I needed to do this to control the AppBar attributes from the page itself, attributes like `leading`, `title`, `centerTitle`, `bottom`, etc...


However, this has downsides:
1. Now I have 2 Scaffolds in my app, one from `AutoTabsScaffold` and the other from my pages

- This caused a bug with Flutter 2.8 while showing `MaterialBanner`, where the banner will extend the appBar
- According to the documentation "it's best to avoid nesting scaffolds" https://api.flutter.dev/flutter/material/Scaffold-class.html

2. The transition animation now includes the `appBar`


I really wish there were someway to delegate the creation of the `appBar` to the pages, while keeping only one Scaffold.
